Formula/git.rb

```diff
@@ -152,6 +152,14 @@ def install
     etc.install "gitconfig"
   end
 
+  def caveats; <<~EOS
+    Sending email via git send-email requires the Perl Net::SMTP::SSL module.
+    If your system Perl does not already have it, you can install it via:
+
+      sudo /usr/bin/cpan Net::SMTP::SSL
```

Comment:
I'm a pretty strong 👎 on telling users to run sudo for system installed stuff, it usually doesn't end well.

Reply:
Me too! That's why I ran into the issue in #33524 in the first place (didn't want to `sudo` anything).

But since the Git fomula does not require perl, and consequently does not use brew Perl's modules, this is the *only* way to make it work.

Reply:
I'm hesitant to add such a specific comment to the 3rd most installed formula. Not to mention the sudo usage. 👎 for me.